File: day4/grid-challenge/solution.py
def gridChallenge(grid):
    splitedSortedArray = []
    for arr in grid:
        arrList = list(arr)
        arrList.sort()
        splitedSortedArray.append(arrList)

    for i in range(len(splitedSortedArray[0])):
        prov = []
        for j in range(len(splitedSortedArray)):
            prov.append(splitedSortedArray[j][i])

        sortedProv = sorted(prov)
        joinedSortedProv = ''.join(sortedProv)
        if (joinedSortedProv != ''.join(prov)):
            return 'NO'

    return 'YES'

File: day4/grid-challenge/solution.js
function gridChallenge(grid) {
  let splitedGrid = grid.map((elem) => elem.split("").sort());
  let columLength = splitedGrid[0].length;
  let lineLength = splitedGrid.length;

  for (let i = 0; i < columLength; i++) {
    let prov = [];
    for (let j = 0; j < lineLength; j++) {
      prov.push(splitedGrid[j][i]);
    }
    let sorted = [...prov].sort().join("");
    if (prov.join("") != sorted) {
      return "NO";
    }
  }
  return "YES";
}
